### ISSUE TITLE
Shared Library Error

### ISSUE THREAD

**Issue by McMattyMatt** [sdetweil/MagicMirror_scripts#65]
The script works flawlessly however on my newest system I keep getting an error while loading shared libraries. It says "cannot open shared object file: No such file or directory" every time I try to launch it

**Comment by sdetweil** [comment]
can u show the rest of the messages?

**Comment by sdetweil** [comment]
is this the just released updated pi os?

**Comment by McMattyMatt** [comment]
It says "/home/user1/MagicMirror/node_modules/electron/dist/electron: error while loading shared libraries: libgobject-2.0.so.0: cannot open shared object file: No such file or directory"

**Comment by McMattyMatt** [comment]
I can still run the shell script in the directory, just not through the terminal

**Comment by McMattyMatt** [comment]
I forgot to mention that's after I deleted electron, for some reason it's having a hard time

**Comment by McMattyMatt** [comment]
Another thing is I am running this on a desktop os pi. Which I've seen users have trouble connecting to electron.

**Comment by sdetweil** [comment]
yes, looks like that file is not installed on x64 systems by default.  just install it

Google search the name for instructions..  

like 
https://pkgs.org/download/libgobject-2.0.so.0

**Comment by McMattyMatt** [comment]
I'm sorry, I'm still having trouble, can you walk me through the terminal process

**Comment by ManuZz80** [comment]
> I'm sorry, I'm still having trouble, can you walk me through the terminal process

You said before: 
I forgot to mention that's after I deleted electron, for some reason it's having a hard time

Maybe you have to install electron again before it could work/find the file …

**Comment by McMattyMatt** [comment]
It says libglib2.0-dev is already the newest version

**Comment by sdetweil** [comment]
sorry, no idea.  it's an electron/os problem

u can run mm in server mode and use chrome or firefox or Midori...

**Comment by McMattyMatt** [comment]
How would I run it in server mode?

**Comment by sdetweil** [comment]
if you are running pm2(setup from my script) to launch, then edit 
~/MagicMirror/installers/mm.sh
and change

npm start 
to
npm run server

the pm2 restart all

then use browser to connect to localhost:8080
(defaults) in config.js

**Comment by sdetweil** [comment]
any update?

**Comment by Griefed** [comment]
I encountered the same issue. Your suggested fix of editing `MagicMirror/installers/mm.sh` and changing `npm start` to `npm run server` followed by `pm2 restart MagicMirror` resolved this for me.

Thank you very much for your hard work!

Cheers,
Griefed
